### PR TITLE
fix: avoid division by zero in GEval score normalization

### PR DESCRIPTION
**Pull request by aerosta**
## Summary

This fixes a division-by-zero edge case in `GEval` score normalization.

When `score_range` has no span, for example `(5, 5)` or `(1, 1)`, the normalization formula divides by `self.score_range_span`, which raises `ZeroDivisionError`. This affects both normalization paths in `g_eval.py`.

## Changes

- added a zero-span guard to both normalization branches
- when `score_range_span == 0`, the code now falls back to `int(g_score)` instead of dividing by zero

## Tests

Added `tests/test_metrics/test_g_eval_score_normalization.py` to cover:

- standard normalization behavior
- custom score range normalization
- zero-span edge cases such as `(5, 5)` and `(1, 1)`
- strict mode behavior
- `get_score_range` utility behavior

```bash
python3.11 -m pytest tests/test_metrics/test_g_eval_score_normalization.py -v

### CODE REVIEW

File: tests/test_metrics/test_g_eval_score_normalization.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+"""Tests for GEval score normalization with zero-span score ranges."""
```
Comment (A-Vamshi) [inline tests/test_metrics/test_g_eval_score_normalization.py]: Hey @aerosta, this PR is good, was just wondering if you could move this test inside the `tests/test_metrics/test_g_eval_metric.py` file? Just append these tests at the bottom please, thank you :)

PS: Do you think this problem exists for the `ConversationalGEval` metric as well? Could you see if that's the case and help with the changes there as well? Thank you!
